as/thrift.js

```diff
@@ -41,17 +41,14 @@ function TChannelAsThrift(opts) {
 
     var self = this;
 
-    assert(opts && typeof opts.source === 'string',
-        'must pass source as an argument');
-
     self.thriftSource = opts.source;
-    self.thriftFileName = opts.thriftFileName || 'service.thrift';
+    self.thriftFileName = path.basename(opts.entryPoint || 'service.thrift');
```

Comment:
The thriftFileName is only retained here, using the basename only, to keep some tests passing that depend on it. It is not otherwise used since we call into self.spec.getSources() to produce the ThriftIDLs bundle.